mission/functions/systems/vehicle_saving/fn_veh_load.sqf

```diff
@@ -0,0 +1,79 @@
+/*
+    File: fn_veh_load.sqf
+    Author: Ridderrasmus
+    Public: No
+
+    Description:
+	    Loads saved vehicle data.
+
+    Parameter(s):
+        _vehData - The data to load [ARRAY]
+
+    Returns: nothing
+
+
+    Example(s):
+	    ["_id", "_class", "_loc", "_data", "_dataCargo", ["_packaged", false]] call vn_mf_fnc_veh_load;
+*/
+
+params ["_vehData"];
+
+_vehData params ["_id", "_class", "_loc", "_data", "_dataCargo", ["_packaged", false]];
+
+/* ONLY WORKS ONCE DOUBLE CHECKING IN MAIN MIKE FORCE IS IMPLEMENTED
+_veh = [_id] call vn_mf_fnc_veh_asset_get_by_id select struct_veh_asset_info_m_vehicle;
+*/
+
+_vehicles = vehicles select {typeOf _x == _class && !(_x getVariable ["rid_loaded", false])};
+//_vehicles = _vehicles select {_x call compile preprocessFileLineNumbers "rid_save_sys\rid_fn_area_check.sqf"};
+if (count _vehicles >= 1) then {
+	_veh = (_vehicles select 0);
+	_veh setVariable ["rid_loaded", true, true];
+	_id call vn_mf_fnc_veh_asset_unlock_vehicle;
+
+
+	// Load vehicle position and direction
+	_veh setPos (_loc select 0);
+	_veh setDir (_loc select 1);
```

Comment:
`setPos` format is not compatible `getPos` and will cause issues if vehicle is over and object. `setDir` should also be avoided as it resets up vector.

Use position world and vectorDirAndUp